### ISSUE TITLE
Recursively self-referential generic Java class causes MonomorphizeClass to not terminate

### ISSUE THREAD

**Issue by wandernauta** [utwente-fmt/vercors#1279]
The following class is accepted by `javac`, but causes the VerCors `monomorphizeClass` rewrite to recurse forever:

```java
class A<T> {
    A<A<T>> walker;
}
```

VerCors appears to hang until it crashes with a stack overflow exception. The same occurs when the snippet is parsed as PVL.

It's probably unlikely that someone would actually need or write such a class, but I would still not expect VerCors to get stuck - instead, I would expect an error message explaining that the class can not be monomorphized.

**Comment by bobismijnnaam** [comment]
I agree with your conclusion. Not sure if the added complexity of the check is proportional to the benefit of a nice error message, though... (Given the rarity of using generics like this)
